tests/test_json_generator.py

```diff
@@ -0,0 +1,177 @@
+# pypam-based-processing
+# Filename: tests/test_json_generator.py
+# Description:  Test fixtures for the json generator classes.
+# Tests the ability to generate metadata for soundtrap, iclisten, and nrs recording files.
+
+import json
+
+import boto3
+import botocore
+import pytest
+from botocore.exceptions import ClientError
+from datetime import datetime
+
+import logging
+
+from pathlib import Path
+
+from src.json_generator.gen_nrs import NRSMetadataGenerator
+from src.logging_helper import create_logger
+from src.json_generator.gen_soundtrap import SoundTrapMetadataGenerator
+from src.json_generator.gen_iclisten import IcListenMetadataGenerator
+
+
+def get_aws_account() -> str:
+    """
+    Get the account number associated with this user
+    :return:
+    """
+    try:
+        account_number = boto3.client('sts').get_caller_identity()['Account']
+        print(f'Found account {account_number}')
+        return account_number
+    except ClientError as e:
+        print(e)
+        msg = f'Could not get account number from AWS. Check your config.ini file. ' \
+              f'Account number is not set in the config.ini file and AWS credentials are not configured.'
+        print(msg)
+        return None
+    except botocore.exceptions.NoCredentialsError as e:
+        print(e)
+        return None
+
+
+# Check if an AWS account is configured by checking if it can access the model with the default credentials
+AWS_AVAILABLE = False
+if get_aws_account():
+    AWS_AVAILABLE = True
+
+
+@pytest.mark.skipif(not AWS_AVAILABLE,
+                    reason="This test is excluded because it requires a valid AWS account")
+def test_soundtrap_json_generator():
```

Comment:
This test is failing for me locally. Though I do have stuff under my `~/.aws` perhaps my credentials have expired.  To move on quickly, I renamed that dir to something else and now this test is skipped (which is good).